packages/common/src/services/coingecko/CoingeckoClient.ts

```diff
@@ -0,0 +1,87 @@
+import { UnixTime } from '../../model/UnixTime'
+import { RateLimiter } from '../../tools/RateLimiter'
+import { HttpClient } from '../HttpClient'
+import { CoingeckoId } from './CoingeckoId'
+import {
+  CoinListEntry,
+  CoinListPlatformEntry,
+  CoinListPlatformResult,
+  CoinListResult,
+  CoinMarketChartRangeData,
+  CoinMarketChartRangeResult,
+} from './model'
+
+const API_URL = 'https://api.coingecko.com/api/v3'
+
+export class CoingeckoClient {
+  private rateLimiter = new RateLimiter({
+    callsPerMinute: 40,
+  })
+  private timeoutMs = 10_000
+
+  constructor(private httpClient: HttpClient) {
+    this.query = this.rateLimiter.wrap(this.query.bind(this))
+  }
+
+  async getCoinList(options?: {
+    includePlatform: false
+  }): Promise<CoinListEntry[]>
+  async getCoinList(options: {
+    includePlatform: true
+  }): Promise<CoinListPlatformEntry[]>
+  async getCoinList(options = { includePlatform: false }) {
+    const data = await this.query('/coins/list', {
+      include_platform: options.includePlatform.toString(),
+    })
+    if (!options.includePlatform) {
+      return CoinListResult.parse(data)
+    } else {
+      return CoinListPlatformResult.parse(data)
+    }
+  }
+
+  async getCoinMarketChartRange(
+    coindId: CoingeckoId,
+    vs_currency: string,
+    from: UnixTime,
+    to: UnixTime
+  ): Promise<CoinMarketChartRangeData> {
+    const data = await this.query(`/coins/${coindId}/market_chart/range`, {
+      vs_currency: vs_currency.toLowerCase(),
+      from: from.toString(),
+      to: to.toString(),
+    })
+
+    const parsedData = CoinMarketChartRangeResult.parse(data)
+
+    return {
```

Comment:
You can actually make it much nicer. You can use proper names instead of `value` and convert timestamps to Dates giving them a more descriptive type than just `number`. You can also use destructuring in the map callback and rely on the inferred type.
```ts
return {
  prices: parsedData.prices.map(([timestamp, price]) => ({
    date: new Date(timestamp),
    price,
  })),
  marketCaps: parsedData.market_caps.map(([timestamp, marketCap]) => ({
    date: new Date(timestamp),
    marketCap,
  })),
  totalVolumes: parsedData.total_volumes.map(([timestamp, totalVolume]) => ({
    date: new Date(timestamp),
    totalVolume,
  })),
}
```